compiler/src/iree/compiler/Dialect/Util/Analysis/Constant/OpOracle.cpp

```diff
@@ -56,8 +56,7 @@ bool isLegalConstExprType(Type t) {
     // support, but for now the consteval JIT has interop limitations. Lift
     // this restriction when the JIT interops for all types.
     auto bitWidth = t.getIntOrFloatBitWidth();
-    return bitWidth == 1 || bitWidth == 8 || bitWidth == 16 || bitWidth == 32 ||
-           bitWidth == 64;
+    return llvm::isPowerOf2_64(bitWidth) && bitWidth != 2 && bitWidth <= 64;
```

Comment:
Thinking about this now, I think this may be okay as long as we don't hoist any i4 constants. I'm not too familiar with the const eval flow, so I don't know if this by itself will need support for subbyte single element types (or bitcasting) in the runtime. I'll let others weigh in on this.

As far as extending it to `i2`, I don't think we even have support for `i2` inside of dispatches yet. We should keep it restricted to just `i4` until we have that. I think there are a lot of places in subbyte emulation where we explicitly check for `i4` (not even Po2), so there will likely be quite a but of work to do making sure `i2` can be emulated as well.

Reply:
Fixed.